prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/application/DeployApplicationTask.java

```diff
@@ -0,0 +1,132 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.redhat.parodos.workflow.context.WorkContextDelegate;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.openshift.client.OpenShiftClient;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@AllArgsConstructor
+public class DeployApplicationTask extends BaseWorkFlowTask {
+
+	public static final String KUBECONFIG = "kubeconfig";
+
+	public static final String MANIFESTS_PATH = "manifestsPath";
+
+	public static final String NAMESPACE = "namespace";
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key(KUBECONFIG).type(WorkParameterType.TEXT).optional(false)
+						.description("kubeconfig file of the target cluster in json format").build(),
+				WorkParameter.builder().key(MANIFESTS_PATH).type(WorkParameterType.TEXT).optional(false)
+						.description("The path to the manifests to deploy the application").build(),
+				WorkParameter.builder().key(NAMESPACE).type(WorkParameterType.TEXT).optional(false)
+						.description("The namespace in which the application should be deployed").build());
+	}
+
+	/**
+	 * Executes the task logic, based on the following steps:
+	 * <ol>
+	 * Verify parameters are present
+	 * </ol>
+	 * <ol>
+	 * Load the manifests
+	 * </ol>
+	 * <ol>
+	 * Load the kubeconfig file and prepare the client
+	 * </ol>
+	 * <ol>
+	 * Deploy the manifests
+	 * </ol>
+	 * @param workContext context in which this unit of work is being executed
+	 * @return a {@link WorkReport} with the result of the execution
+	 */
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		final String kubeconfig;
+		final String manifestsPath;
+		final String namespace;
+
+		// Verify parameters are present
+		try {
+			kubeconfig = WorkContextDelegate.getRequiredValueFromRequestParams(workContext, KUBECONFIG);
+			manifestsPath = WorkContextDelegate.getRequiredValueFromRequestParams(workContext, MANIFESTS_PATH);
+			namespace = WorkContextDelegate.getRequiredValueFromRequestParams(workContext, NAMESPACE);
+		}
+		catch (MissingParameterException e) {
```

Comment:
There is an ilelligalParameter exception if I recall correctly

Reply:
https://github.com/parodos-dev/parodos/blob/d5eab829844b152d6f3d0e398fb23c10857a430c/parodos-model-api/src/main/java/com/redhat/parodos/workflow/context/WorkContextDelegate.java#L181

```
	public static String getRequiredValueFromRequestParams(WorkContext workContext, String key)
			throws MissingParameterException {
```